### PR TITLE
backend: (JAX) add export to JAX executable

### PR DESCRIPTION
**Pull request by superlopuh**
PR 1/2, allowing for the creation of executable objects, with an `execute` method that just calls the JAX underlying executable.

### CODE REVIEW

File: xdsl/backend/jax_executable.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    Usage:
+
+    ``` python
+    my_module_str =\"""
+    func.func @main(%arg0: tensor<5xf32>) -> tensor<5xf32> {
+        ...
+    }
+    \"""
+    my_module = parser.parse_module(my_module_str)
+    @JaxExecutable.compile(module)
```
Comment (zero9178) [inline xdsl/backend/jax_executable.py:50]:
```suggestion
    @JaxExecutable.compile(my_module)
```

Comment (zero9178) [inline xdsl/backend/jax_executable.py:50]: I don't see a unit test using the decorator syntax nor a `__call__` that would delegate to `execute`. Is this intentional?

Comment (zero9178) [reply xdsl/backend/jax_executable.py:50]: Just saw the stacked PR which implements this. The documentation seems to just be ahead of its time :slightly_smiling_face:

**Review by zero9178 — APPROVED** [review]

File: xdsl/backend/jax_executable.py (hunk 1 above)
Comment (superlopuh) [reply xdsl/backend/jax_executable.py]: 🔮

Comment (superlopuh) [reply xdsl/backend/jax_executable.py]: I'll wait for both to be approved to merge this one

**Pull request merged**